src/core/tsi/alts/crypt/aes_gcm.cc

```diff
@@ -182,38 +182,87 @@ static void aes_gcm_mask_nonce(uint8_t* dst, const uint8_t* nonce,
   memcpy(dst + sizeof(nonce1), &nonce2, sizeof(nonce2));
 }
 
-static grpc_status_code aes_gcm_derive_aead_key(uint8_t* dst,
-                                                const uint8_t* kdf_key,
-                                                const uint8_t* kdf_counter) {
-  unsigned char buf[EVP_MAX_MD_SIZE];
-  unsigned char ctr = 1;
+grpc_status_code aes_gcm_derive_data(uint8_t* out_key, size_t out_size,
```

Comment:
Does the name `aes_gcm_derive_secret` work? `data` might be a bit misleading.

Reply:
Changed the name to `hkdf_derive_secret`, since it really isn't AES-GCM specific.